File: src/025. 1000-digit Fibonacci number/025.java
import java.math.*;

public class Main {
    
    public static void main(String[] args) {
        BigInteger fa = BigInteger.ONE;
        BigInteger fb = BigInteger.ONE;
        
        int index = 1;
        while (fa.toString().length() < 1000) {
            BigInteger temp = fb;
            fb = fa.add(fb);
            fa = temp;
            index++;
        }
        System.out.println(index);
    }
}

File: src/025. 1000-digit Fibonacci number/025.go
package main

import (
	"fmt"
	"math/big"
)

func main() {
	index := 1
	fa, fb := big.NewInt(1), big.NewInt(1)
	var upperLimit big.Int
	upperLimit.Exp(big.NewInt(10), big.NewInt(999), nil)
	for fa.Cmp(&upperLimit) < 0 {
		var temp big.Int
		temp.Set(fb)
		fb.Add(fa, fb)
		fa.Set(&temp)
		index++
	}
	fmt.Println(index)
}
